hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -686,3 +806,261 @@ func (r *NodePoolReconciler) listAWSMachineTemplates(nodePool *hyperv1.NodePool)
 	}
 	return filtered, nil
 }
+
+func reconcileMCSDeployment(nodePool *hyperv1.NodePool, deployment *appsv1.Deployment, sa *corev1.ServiceAccount, releaseImage *releaseinfo.ReleaseImage) error {
+	images := releaseImage.ComponentImages()
+	targetVersion := releaseImage.Version()
+	// We use the nodePool name as the label to be selected by the Service.
+	appName := nodePool.GetName()
+
+	bootstrapArgs := fmt.Sprintf(`
```

Comment:
Here's a place where I think an inline template might actually be valuable 😁  but that's just a note for some other time, no change requested for this PR.